spec/palletjack-pallet_spec.rb

```diff
@@ -1,45 +1,53 @@
 require 'spec_helper'
 require 'rspec_structure_matcher'
+require 'traceable'
 
 describe PalletJack::Pallet do
   before :context do
     @jack = PalletJack.load($EXAMPLE_WAREHOUSE)
   end
-  
+
   it 'can load contents of a pallet' do
     expect {
       path = File.join('domain', 'example.com')
       identity = PalletJack::Pallet::Identity.new(@jack, path)
       PalletJack::Pallet.load(@jack, identity)
     }.not_to raise_error
   end
-  
+
   it 'cannot load contents of a non-existent pallet' do
     expect {
       path = File.join('__invalid__', '__no_such__')
       identity = PalletJack::Pallet::Identity.new(@jack, path)
       PalletJack::Pallet.load(@jack, identity)
     }.to raise_error Errno::ENOENT
   end
-  
+
   context 'a loaded pallet' do
     before :context do
       path = File.join('domain', 'example.com')
       identity = PalletJack::Pallet::Identity.new(@jack, path)
       @pallet = PalletJack::Pallet.load(@jack, identity)
     end
-    
+
     it 'has a kind' do
       expect(@pallet.kind).to match 'domain'
     end
-    
+
     it 'has a name' do
       expect(@pallet.name).to match 'example.com'
     end
-    
+
     it 'serializes into YAML of its key contents' do
       pallet_hash = @pallet.to_hash
-      yaml_hash = YAML::load(@pallet.to_yaml)
+
+      handler = PositionHandler.new('@pallet')
+      parser = Psych::Parser.new(handler)
+      handler.parser = parser
+      parser.parse(@pallet.to_yaml)
+      visitor = PositionVisitor.new
+      yaml_hash = visitor.accept(handler.root)[0]
+
```

Comment:
Because `YAML::load` loads integers as integers, while the pallet has them as `TraceableString`:s.